channeldb/migration13/migration.go

```diff
@@ -0,0 +1,150 @@
+package migration13
+
+import (
+	"fmt"
+
+	"github.com/coreos/bbolt"
+)
+
+var (
+	paymentsRootBucket = []byte("payments-root-bucket")
+
+	// paymentFailInfoKey is a key used in the payment's sub-bucket to
+	// store information about the reason a payment failed.
+	paymentFailInfoKey = []byte("payment-fail-info")
+
+	// paymentAttemptInfoKey is a key used in the payment's sub-bucket to
+	// store the info about the latest attempt that was done for the
+	// payment in question.
+	paymentAttemptInfoKey = []byte("payment-attempt-info")
+
+	// paymentSettleInfoKey is a key used in the payment's sub-bucket to
+	// store the settle info of the payment.
+	paymentSettleInfoKey = []byte("payment-settle-info")
+
+	// paymentHtlcsBucket is a bucket where we'll store the information
+	// about the HTLCs that were attempted for a payment.
+	paymentHtlcsBucket = []byte("payment-htlcs-bucket")
+
+	// htlcAttemptInfoKey is a key used in a HTLC's sub-bucket to store the
+	// info about the attempt that was done for the HTLC in question.
+	htlcAttemptInfoKey = []byte("htlc-attempt-info")
+
+	// htlcSettleInfoKey is a key used in a HTLC's sub-bucket to store the
+	// settle info, if any.
+	htlcSettleInfoKey = []byte("htlc-settle-info")
+
+	// htlcFailInfoKey is a key used in a HTLC's sub-bucket to store
+	// failure information, if any.
+	htlcFailInfoKey = []byte("htlc-fail-info")
+)
+
+func MigrateMPP(tx *bbolt.Tx) error {
+	// Iterate over all payments.
+	paymentsBucket := tx.Bucket(paymentsRootBucket)
+	if paymentsBucket == nil {
+		return nil
+	}
+
+	return paymentsBucket.ForEach(func(k, v []byte) error {
+		bucket := paymentsBucket.Bucket(k)
+
+		// We only expect sub-buckets to be found in
+		// this top-level bucket.
+		if bucket == nil {
+			return fmt.Errorf("non bucket element in " +
+				"payments bucket")
+		}
+
+		// No migration needed if there is no attempt stored.
+		attemptInfo := bucket.Get(paymentAttemptInfoKey)
+		if attemptInfo == nil {
+			return nil
+		}
+
+		// Delete attempt info on the payment level.
+		if err := bucket.Delete(paymentAttemptInfoKey); err != nil {
+			return err
+		}
+
+		// Save attempt id for later use.
+		attemptID := attemptInfo[:8]
+
+		// Discard attempt id. It will become a bucket key in the new
+		// structure.
+		attemptInfo = attemptInfo[8:]
+
+		// Append unknown (zero) attempt time.
+		attemptInfo = append(
+			attemptInfo, 0, 0, 0, 0, 0, 0, 0, 0,
```

Comment:
idt this is the correct serialization, is this equivalent to marshaled `time.Time` values that include time-zone?

Reply:
We serialize unix nano seconds and those don't include a timezone. They are always expressed in UTC. I think the zeroes here are correct, but I did extend the serialize/deserialize methods to convert between zero `time.Time` and zero unix nano seconds.

Reply:
Resolved, not storing tz